api/types/types.proto

```diff
@@ -2239,3 +2239,38 @@ message LockTarget {
     // MFADevice specifies the UUID of a user MFA device.
     string MFADevice = 6 [ (gogoproto.jsontag) = "mfa_device,omitempty" ];
 }
+
+// AddressCondition represents a set of addresses. Presently the addresses are specfied
+// exclusively in terms of IPv4/IPv6 ranges.
+message AddressCondition {
+    // CIDR is IPv4 or IPv6 address. Valid value are either CIDR ranges (e.g. "10.0.1.0/24",
+    // "fe::/8") or a single IP address (e.g "10.1.2.3")
+    string CIDR = 1 [ (gogoproto.jsontag) = "cidr" ];
+}
+
+message NetworkRestrictionsSpecV4 {
+    // Allow lists the addresses that should be allowed.
+    repeated AddressCondition Allow = 5
+        [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "allow" ];
+    // Deny lists the addresses that should be denied even if they're allowed by Allow condition.
+    repeated AddressCondition Deny = 6
+        [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "deny" ];
```

Comment:
Any reason these are `5` and `6` instead of `1` and `2`?

Reply:
Good catch. There was a previous version with more members. Should be 1 and 2 now.